server/middleware/validationMiddleware.ts

```diff
@@ -83,7 +83,12 @@ export const validate = (schema: z.ZodTypeAny | SchemaFactory): RequestHandler =
   }
 }
 
-export const validateDateEarlierThanToday = (requiredErr: string, invalidErr: string, maxErr: string) => {
+export const validateDate = (
+  requiredErr: string,
+  invalidErr: string,
+  maxErr: string,
+  allowedValues: 'PAST' | 'FUTURE' | 'ALL',
+) => {
```

Comment:
I've split this up into two different  exported methods in my PR, I think its simpler than doing extra logic here with allowedValues. (first change here https://github.com/ministryofjustice/hmpps-challenge-support-intervention-plan-ui/pull/163/files - validateDateNoTransform name could be better tho!)

Reply:
oops sorry yeah noticed and edited before I saw this response! please check above